src/test/java/com/google/jenkins/plugins/k8sengine/KubectlWrapperVerboseLoggingTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jenkins.plugins.k8sengine;
+
+import com.google.common.collect.ImmutableList;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.Launcher.ProcStarter;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for verifying the use of the verboseLogging flag in {@link KubectlWrapper}. */
+@RunWith(MockitoJUnitRunner.class)
+public class KubectlWrapperVerboseLoggingTest {
+  @Test(expected = IOException.class)
+  public void testQuietLoggingApplied() throws IOException, InterruptedException {
+    KubectlWrapper kubectlWrapper =
+        Mockito.spy(
+            new KubectlWrapper.Builder()
+                .launcher(setUpLauncher())
+                .kubeConfig(setUpKubeconfig())
+                .workspace(setUpWorkspace())
+                .namespace("")
+                .verboseLogging(false)
+                .build());
+    kubectlWrapper.runKubectlCommand("apply", ImmutableList.of("-f", "manifest.yaml"));
```

Comment:
runKubectlCommand returns the output as as string. It would be great to see an assertions verifying the different logging levels.

Reply:
The ProcStarter.quiet flag doesn't have an effect on the contents of that output string, it determines whether the command output is printed to the real stdout/stderr. The full output is required for use in verification to get the manifest json. When you provide stdout and stderr for the launch, it seems like it is adding another stream to write the output to rather than replacing the original stream.

Reply:
Rather than testing the behavior of the launch, I opted to make sure that the quiet flag was being set properly based on the kubectl.verboseLogging flag.

It might be possible to do an integration test that checks for this.

Reply:
Discussed this offline. Since this isn't a super critical feature, just testing for the flag should suffice.